code/week5/week5.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * to compile
+ * g++ -std=c++14 -o week5.out *.cpp
+ */
+#include <iostream>
```

Comment:
This isn't compiling for me without

```
#include <memory>
```
to get the defs for unique_ptr and such